Formula/tg.rb

```diff
@@ -0,0 +1,28 @@
+class Tg < Formula
+  include Language::Python::Virtualenv
+
+  desc "Terminal telegram client"
+  homepage "https://github.com/paul-nameless/tg"
+  url "https://github.com/paul-nameless/tg/archive/refs/tags/v0.11.0.tar.gz"
+  sha256 "2303983af7b7246fb3198bcae688d3b6327a96bbfc91c94ef9bc783cc7191f42"
+  license "Unlicense"
+
+  bottle :unneeded
+
+  depends_on "python@3.9"
+  depends_on "tdlib" => "1.7.0"
+
+  resource "python-telegram-patched" do
+    url "https://github.com/paul-nameless/python-telegram/archive/refs/tags/0.14.0-patched.tar.gz"
+    sha256 "7f3af4cb722f37bd65b0a49da448b7d26afa7f2de761aed4828018155e2e78b4"
+  end
+
+  def install
+    virtualenv_install_with_resources
+  end
+
+  test do
+    (testpath/".config/tg/conf.py").write("")
+    assert_match version.to_s, shell_output("#{bin}/tg -v")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
There is no really other way to check it, cause it's TUI app.

Reply:
There are a bunch of other terminal UI apps that do have tests. You could start, wait and stop the app and see if that works without errors.

Reply:
You can also use Ruby's `pty` module to try to interact with the TUI.

Reply:
I've updated the test, that should be enough as it will load the dependency and if they broken it will fail